app/models/in_person_enrollment.rb

```diff
@@ -133,8 +133,9 @@ def on_notification_sent_at_updated
     end
   end
 
-  def skip_notification_sent_at_set?
-    !notification_phone_configuration.present? || (!self.passed? && !self.failed? && !self.expired?)
+  def eligible_for_notification?
+    self.notification_phone_configuration.present? &&
+      (self.passed? || self.failed? || self.expired?)
   end
```

Comment:
The naming here seems more intuitive, but I think it could also be misleading. I would suggest updating this method as such:

```suggestion
  def eligible_for_notification?
    self.notification_phone_configuration.present? &&
      (self.passed? || self.failed?)
  end
```

... then update the `on_status_updated` method as such:

```ruby
  def on_status_updated
    if (enrollment_will_be_cancelled? || enrollment_will_be_expired?) && notification_phone_configuration.present?
      notification_phone_configuration.destroy!
    end
    self.status_updated_at = Time.zone.now
  end

  def enrollment_will_be_expired?
    status_change_to_be_saved&.last == 'expired'
  end

  def enrollment_will_be_cancelled?
    status_change_to_be_saved&.last == 'cancelled'
  end
```

... and finally remove the `if enrollment.expired?` conditional from `SendProofingNotificationJob`.

Reply:
Agreed.. I am planning on doing this in a follow-up PR because it feels like too much of a change for a 'refactor' PR

Reply:
I would be inclined to exclude this method rename/refactor entirely from the current PR if this change will be addressed in a follow-up PR.

Reply:
#8875 is the follow-on PR